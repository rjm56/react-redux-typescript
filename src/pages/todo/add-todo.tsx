import * as React from 'react';

interface IAddToDoProps {

}

export const AddToDo = (props: IAddToDoProps) => {
    return (<div></div>)
}