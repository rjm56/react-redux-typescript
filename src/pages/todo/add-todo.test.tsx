import * as React from 'react';
import { mount } from 'enzyme';
import { AddToDo, IAddToDoProps } from './add-todo';

describe('<AddToDo />', () => {

  it('should render without crashing', () => {

    let props: IAddToDoProps = {
      addTodo: () => {}
    }

    const wrapper = mount(<AddToDo {...props}/>)

    expect(wrapper).toMatchSnapshot()
  });

  it('should call addTodo onClick of <button/>', () => {
    const addTodo = jest.fn();

    let newProps: IAddToDoProps = {
      addTodo
    }

    const wrapper = mount(<AddToDo {...newProps}/>);
    wrapper.find('button').simulate('click');
    expect(addTodo).toHaveBeenCalledTimes(1);
  });
})