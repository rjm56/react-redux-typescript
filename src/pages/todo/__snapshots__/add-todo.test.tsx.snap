// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToDo /> should render without crashing 1`] = `
<AddToDo
  addTodo={[Function]}
>
  <div>
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add todo
    </button>
  </div>
</AddToDo>
`;
